### PR TITLE
ab: access only owner's assignments and courses

### PR DESCRIPTION
**Pull request by annagav**
Filter assignments and courses by owner. 
Also removed two unused views.

### CODE REVIEW

**Comment by noisecapella** [comment]
When I step through the application as a new user I don't see other user's courses or assignments, which is fine, but there are still endpoints which aren't checking ownership properly. For example anyone can go to `/ab/assignments/delete/34` and delete assignment 34 without owning it. `publish_assignment` is also unrestricted.

**Comment by noisecapella** [comment]
By the way if you want to get this PR merged and deal with the other security issues later, that's fine with me

File: instructor/common.py [hunk 1]
```diff
@@ -131,7 +104,7 @@ def course_setup(request):
                 course.save()
 
     # if there is at least one course
-    all_courses = models.Course.objects.all()
+    all_courses = models.Course.objects.filter(owner__username=request.user)
```
Comment (noisecapella) [inline instructor/common.py]: It seems like this should compare against `request.user.username`

**Comment by annagav** [comment]
Let's do the rest in another PR :)

**Comment by noisecapella** [comment]
:+1:

**Pull request merged**